### PR TITLE
Add ApiError to RateLimitException to access the response body

### LINKED ISSUE #479: RateLimitApiException doesn't have response body content
### Describe the problem you'd like to have solved

The RateLimitApiException doesn't contain the whole information given by the Auth0's Authentication API. The RateLimitApiException is thrown on every single HTTP 429 response, and, currently only contains the rate limit information - from the response's header. 
The Authentication API will also return HTTP 429 in other situations like when the Brute Force feature is enabled on tenant and an user has failed to login more than 10 times (Resource Owner Password flow). Please see the screenshot below
image
The "error" and "error_description" properties aren't available in RateLimitApiException.

### Describe the ideal solution

The RateLimitApiException should include data contained in response body - not only what is in the headers.

## Alternatives and current work-arounds

No alternatives have been found. The error handling factory cannot be replaced in the SDK

### PR DESCRIPTION
**Pull request by fernandozpiccin**
### Changes

- Added `ApiError` as a property of `RateLimitApiException` so the response body can be accessible when a 429 HTTP response code is returned by the APIs e.g. when IP address is blocked for an specific user account due to Brute Force Protection feature being enabled

### References

Closes #479

### Testing

Please describe how this can be tested by reviewers. Be specific about anything not tested and reasons why. If this library has unit and/or integration testing, tests should be added for new functionality and existing tests should complete without errors. 

- [x] This change adds unit test coverage

- [ ] This change adds integration test coverage
Changes were made in core. Unit tests cover these changes made.

- [ ] This change has been tested on the latest version of the platform/language or why not
Integration tests weren't ran. Not sure if that required a special tenant to be configured in advance. Can you please run that for me or provide additional information on how to set-up a tenant for integration testing running?

### Checklist

- [x] I have read the [Auth0 general contribution guidelines](https://github.com/auth0/open-source-template/blob/master/GENERAL-CONTRIBUTING.md)

- [x] I have read the [Auth0 Code of Conduct](https://github.com/auth0/open-source-template/blob/master/CODE-OF-CONDUCT.md)

- [x] All existing and new tests complete without errors

### CODE REVIEW

File: src/Auth0.Core/Exceptions/RateLimitApiException.cs [hunk 1]
```diff
@@ -46,15 +52,17 @@ public RateLimitApiException(string message, Exception innerException)
         /// Initializes a new instance of the <see cref="RateLimitApiException"/> class with a specified <paramref name="rateLimit"/>.
         /// </summary>
         /// <param name="rateLimit"><see cref="Exceptions.RateLimit"/> received on the API call that failed.</param>
-        public RateLimitApiException(RateLimit rateLimit)
+        /// <param name="apiError"><see cref="Exceptions.ApiError"/> received on the API call that failed.</param>
+        public RateLimitApiException(RateLimit rateLimit, ApiError apiError)
```
Comment (frederikprijck) [inline src/Auth0.Core/Exceptions/RateLimitApiException.cs:56]: In order to stay backwards compatible, this constructor should take an optional ApiError.
Even though a user is not expected to be instantiating this class themselves, I think it's a good idea to not break existing APIs when not neccessary.

**Review by frederikprijck — CHANGES_REQUESTED** [review]
Thanks for the PR. I will worry about ensuring the integration tests work.

Added one comment to ensure backwards compatibility.

File: src/Auth0.Core/Exceptions/RateLimitApiException.cs (hunk 1 above)
Comment (fernandozpiccin) [reply src/Auth0.Core/Exceptions/RateLimitApiException.cs:56]: Good point. We cannot assume that nobody else is using that constructor.
I'll make the changes in a second.
re integration tests: Does it should run against any auth0 tenants? Or does it require a tenant with specific config? I couldn't find the instructions of how to set that up. Thanks

Comment (frederikprijck) [reply src/Auth0.Core/Exceptions/RateLimitApiException.cs]: >re integration tests: Does it should run against any auth0 tenants? Or does it require a tenant with specific config? I couldn't find the instructions of how to set that up. Thanks

Ye it needs some enterprise setup that not everyone is able to set up.

**Comment by frederikprijck** [comment]
@fernandozpiccin unless something comes up with the implementation of this PR, I aim to get this tested, merged and released somewhere tomorrow.

**Review by frederikprijck — APPROVED** [review]
Thanks! 🚀

**Pull request merged**